File: Easy/203. Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        preHead = ListNode(0)
        preHead.next = head
        
        preNode = preHead
        while preNode.next:
            if preNode.next.val == val:
                preNode.next = preNode.next.next
            else:
                preNode = preNode.next

        return preHead.next

File: Easy/203. Remove Linked List Elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *removeElements(ListNode *head, int val) {
        ListNode *preHead = new ListNode(0);
        preHead->next = head;

        ListNode *preNode = preHead;
        while(preNode->next){
            if(preNode->next->val == val){
                preNode->next = preNode->next->next;
            }else{
                preNode = preNode->next;
            }
        }
        return preHead->next;
    }
};